File: core/core-api/src/main/java/com/coffee/gu/api/controller/v1/response/stamp/StampHistoryResponse.java
package com.coffee.gu.api.controller.v1.response.stamp;

import com.coffee.gu.stamp.StampHistory;
import com.coffee.gu.stamp.StampHistoryItem;

import java.util.List;

public record StampHistoryResponse(
        long expiringSoonCount,
        List<StampHistoryItem> histories
) {
    public static StampHistoryResponse of(long expiringSoonCount, List<StampHistory> stampHistories) {
        List<StampHistoryItem> stampHistoryItems = stampHistories.stream()
                .map(StampHistoryItem::from)
                .toList();
        return new StampHistoryResponse(expiringSoonCount, stampHistoryItems);
    }


}

File: core/core-api/src/main/kotlin/com/coffee/gu/api/controller/v1/response/stamp/StampHistoryResponse.kt
package com.coffee.gu.api.controller.v1.response.stamp

import com.coffee.gu.stamp.StampHistory
import com.coffee.gu.stamp.StampHistoryItem

class StampHistoryResponse(
    val expiringSoonCount: Int,
    val histories: List<StampHistoryItem>,
) {
    companion object {
        @JvmStatic
        fun of(expiringSoonCount: Int, stampHistories: List<StampHistory>): StampHistoryResponse {
            val stampHistoryItems = stampHistories.map { StampHistoryItem.from(it) }
            return StampHistoryResponse(expiringSoonCount, stampHistoryItems)
        }
    }
}
